### PR TITLE
add support for legacy modules and dynamic config

### PR DESCRIPTION
**Pull request by phated**
Relies on https://github.com/tkellen/node-interpret/pull/15

This is my POC for how the config objects would work.  It definitely needs to be cleaned up but all tests are passing and they even pass with coffee-script < 1.7 (will need to add tests for that).

### CODE REVIEW

**Comment by phated** [comment]
I like that better than my implementation.  I wasn't liking how the cleanliness of the extensions list was lost.  I will try to update this soon.

**Comment by phated** [comment]
pinging due to new changes. let me know your thoughts @silkentrance @tkellen

**Comment by tkellen** [comment]
Can you update this to depend on `interpret@0.4.0`?  Also, any thoughts on #7?

**Comment by tkellen** [comment]
Whoops, I mean #10

**Comment by phated** [comment]
I've struggled with the same problem that is described in the past with no good resolution.  Passing module down is quite scary in my opinion.  Is `load` necessary to keep as part of this API?  I prefer the `registerFor` and actually using `require` since it is more explicit.

**Comment by phated** [comment]
Updated, tests passing on travis and even added different coffee-script versions to the travis matrix

File: index.js [hunk 1]
```diff
@@ -17,12 +35,20 @@ exports.registerFor = function (filepath, cwd) {
     cwd = path.dirname(path.resolve(filepath));
```
Comment (tkellen) [inline index.js]: this block can be moved below the moduleName lookup, it won't be needed unless a module exists.

**Comment by tkellen** [comment]
:+1: to squash and merge after that minor nit

**Comment by phated** [comment]
Cleaned up and squashed, waiting for travis to run

**Comment by phated** [comment]
Travis passed, merging

**Pull request merged**